crates/hyperqueue/src/client/commands/submit/defs.rs

```diff
@@ -162,14 +148,64 @@ pub struct TaskDef {
     pub crash_limit: u32,
 }
 
+#[derive(Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct TaskDef {
+    pub id: Option<JobTaskId>,
+
+    #[serde(flatten)]
+    pub config: TaskConfigDef,
+}
+
+fn deserialize_array<'de, D>(deserializer: D) -> Result<IntArray, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let buf = String::deserialize(deserializer)?;
+    parse_array(&buf).map_err(serde::de::Error::custom)
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct ArrayDef {
+    #[serde(deserialize_with = "deserialize_array")]
+    pub ids: IntArray,
+
+    #[serde(flatten)]
+    pub config: TaskConfigDef,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct JobDef {
+    #[serde(default)]
+    pub name: String,
+
+    pub max_fails: Option<JobTaskCount>,
+
+    #[serde(default)]
+    #[serde(rename = "task")]
+    pub tasks: Vec<TaskDef>,
```

Comment:
Shouldn't we define these mutually exclusive options with enums? To avoid the manual errors checks. Or did you do it to provide better error messages?